### PR TITLE
Feature/customer functionality

### PR DESCRIPTION
**Pull request by LKessell**
## Reason for Pull Request:
Add functionality for Customer class

## Testing Method(s)
[x] Terminal/Mocha
[ ] localhost
[ ] Lighthouse

## Changes Implemented:
- Add methods to get and sort bookings
- Get past and future bookings
- Calculate the amount spent on rooms

### CODE REVIEW

File: src/Customer.js [hunk 1]
```diff
@@ -1,3 +1,69 @@
...
+    this.id = customerData.id;
+    this.name = customerData.name;
+    this.bookings = [];
+
+    this.getBookings(bookingData);
+  }
+
+  getBookings(bookingData) {
+    const matchedBookings = bookingData.filter(booking => booking.userID === this.id);
+    this.bookings = matchedBookings;
```
Comment (k-atwhite) [inline src/Customer.js:12]: solid

File: src/Customer.js [hunk 2]
```diff
@@ -1,3 +1,69 @@
...
+  }
+
+  sortBookings() {
+    return this.bookings.sort((a, b) => new Date(a.date) - new Date(b.date));
+  }
```
Comment (k-atwhite) [inline src/Customer.js:17]: dunno if you care - but I'm trying to do a past bookings a future bookings... now that i think more about it sounds hard

File: src/Customer.js [hunk 3]
```diff
@@ -1,3 +1,69 @@
...
+
+  getPastBookings(inputDate) {
+    const results = this.bookings.filter(booking => {
```
Comment (k-atwhite) [inline src/Customer.js:20]: oh look... functionality for it...

File: test/customer-test.js [hunk 4]
```diff
@@ -131,12 +130,12 @@ describe('Customer Class', () => {
   });
 
   it('Should calculate the total amount spent on bookings', () => {
-    expect(customer.getTotalSpent()).to.equal(1260.75);
+    expect(customer.getTotalSpent(ledger.rooms)).to.equal(1260.75);
   });
 
   it('Should return 0 if customer has no bookings', () => {
-    const newCustomer = new Customer({ 'id': 6, 'name': 'Bennett Schroeder' });
+    const newCustomer = new Customer({ 'id': 6, 'name': 'Bennett Schroeder' }, ledger.bookings);
```
Comment (k-atwhite) [inline test/customer-test.js:137]: great guest. love bennett

**Review by k-atwhite — COMMENTED** [review]
Solid! I will probably be asking for advice on getting those past/ future dates

**Comment by LKessell** [comment]
@niksseif I hope this is the correct tagging procedure!

File: test/customer-test.js (hunk 4 above)
Comment (LKessell) [reply test/customer-test.js:137]: I was SO hoping you'd notice :)

File: src/Customer.js (hunk 3 above)
Comment (LKessell) [reply src/Customer.js:20]: My functions for the past/future bookings are really similar... If you have any suggestions for combining them I would love to hear! Or maybe a good opportunity for future refactoring.

**Pull request merged**